File: C++/프로그래머스/0/120864. 숨어있는 숫자의 덧셈 （2）/숨어있는 숫자의 덧셈 （2）.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string my_string) {
    int answer = 0;
    int temp = 0;
    
    for(const auto c: my_string) {
        if (c >= '0' && c <= '9') {
            temp *= 10;
            temp += c - '0';
        }
        else if (temp != 0) {
            answer += temp;
            temp = 0;
        }
    }
    
    if (temp != 0) answer += temp;
    
    return answer;
}

File: Go/프로그래머스/0/120864. 숨어있는 숫자의 덧셈 （2）/숨어있는 숫자의 덧셈 （2）.go
func solution(my_string string) int {
    var res, temp int
    
    for _, r := range my_string {
        if r >= '0' && r <= '9' {
            temp *= 10
            temp += int(r - '0')
        } else if temp != 0 {
            res += temp
            temp = 0
        }
    }
    
    if temp != 0 {
        res += temp
    }
    
    return res
}